### PR TITLE
Fix some problems with examples 

### PR DESCRIPTION
**Pull request by darakshan**
Example input files are now read from `example-data/input` and ouput files are now written to `example-data/output`.  The directory is created if necessary.
Image dependency in `Cargo.toml` now includes `.png` support.

### CODE REVIEW

File: palette/Cargo.toml [hunk 1]
```diff
@@ -64,7 +64,7 @@ default-features = false
 
 [dev-dependencies.image]
 version = "0.22"
-default-features = false
+default = ["png_codec"]
```
[the remark this answers is not in the document]
Comment (okaneco) [reply palette/Cargo.toml:67]: The most recent version of image is the [0.23 release](https://github.com/image-rs/image/releases/tag/v0.23.0) which had some interface changes. The `png_codec` feature was replaced with `png`. 
```toml
[dependencies.image]
version = "0.23"
default-features = false
features = ["png"]
```

I don't know what other adjustments are needed in the examples if the dependency is updated to 0.23.

[the remark this answers is not in the document]
Comment (Ogeon) [reply palette/Cargo.toml:67]: Ah, that's right. I forgot there's a new version. @darakshan, you can choose if you want to try upgrading `image` too, but don't pour all your energy into it if it seem to require a lot of rework.

[the remark this answers is not in the document]
Comment (darakshan) [reply palette/Cargo.toml]: corrected. upgrade appears to work with no issues.

**Comment by Ogeon** [comment]
Excellent! But it does still look like `example-data/output/hue.png` is in there. It came with your first commit. You will have to delete it again in a new commit and/or rebase them into a single commit that doesn't contain the file.

**Comment by darakshan** [comment]
Doh! I should have removed it before I changed the .gitignore file.  
I'll take care of it tomorrow...

**Comment by Ogeon** [comment]
Alright, this looks good to be merged to me. :+1:

I'm usually asking to have fixup commits squashed into the main commits after review. Mostly to keep the history a bit easier to follow. In this case it would also be a good way to avoid bringing the output file into the master branch, as it's still a part of the first commit at the moment. Would you mind doing that as a final step?

**Comment by darakshan** [comment]
Just checking in... This has been ready to merge for a few weeks, squashed after your last request. :-)

**Comment by Ogeon** [comment]
Oh, so it has! I must have missed the email I usually get when there are changes. Or maybe it wasn't sent out as it should. Either way, let's get it merged! Thanks a lot for fixing this!

bors r+

**Pull request merged**